src/common/dns.ts

```diff
@@ -0,0 +1,25 @@
+import ipfs from '../ipfs'
+
+
+export function lookupTxtRecord(host: string): Promise<string> {
+  return fetch(`https://cloudflare-dns.com/dns-query?name=${host}&type=TXT`, {
+    headers: {
+      "accept": "application/dns-json"
+    }
+  })
+  .then(r => r.json())
+  .then(r => r.Answer[0].data)
+}
+
+
+export async function lookupDnsLink(host: string): Promise<string> {
+  let t
+
+  try {
+    t = await ipfs.dns(host)
+  } catch (_) {
+    t = await lookupTxtRecord(host)
+  }
```

Comment:
@dholms I've extracted the api and dns stuff. Also, I've used Cloudflare as a backup for the ipfs-dnslink lookup here ☝️ (in case ipfs.io is down again)